src/tokenizer/index.js

```diff
@@ -678,7 +687,7 @@ export default class Tokenizer extends LocationParser {
     } else {
       val = parseInt(str, 8);
```

Comment:
None of these parseInt calls are appropriate if isBigInt, as they will round away part of the answer.